### PR TITLE
Fix: Floating Buttons - Add Playwright Testing 

### PR DESCRIPTION
**Pull request by nicoladj77**
## PR Checklist

- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Add automated testing. I tested the changes we made to the editor ( removing navigator, custom advanced settings tab...), and the correct functioning of set as entire site and click tracking. I have not tested the function of all controls because that fills a little bit too much but let me know. 

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [x] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: modules/floating-buttons/assets/js/shared/frontend/handlers/click-tracking.js [hunk 1]
```diff
@@ -66,7 +66,7 @@ export default class ClickTrackingHandler extends Base {
 
 		this.clicks.push( documentId );
 
-		if ( this.clicks.length >= 10 ) {
+		if ( this.clicks.length >= 2 ) {
```
Comment (nicoladj77) [inline modules/floating-buttons/assets/js/shared/frontend/handlers/click-tracking.js:69]: @louiswol94 this is for testing purposes, maybe should I put it under control of a querystring parameter?

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+		await expect( sections ).toHaveCount( 5 );
+
+		const addNewWidget = editor.getPreviewFrame().locator( EditorSelectors.addNewSection );
+		await expect( addNewWidget ).toBeHidden();
+		const container = editor.getPreviewFrame().locator( EditorSelectors.container );
+		await expect( container ).toBeVisible();
+		const contactButtonWidget = container.locator( EditorSelectors.getWidgetByName( 'contact-buttons' ) );
+		await expect( contactButtonWidget ).toBeVisible();
+		const contactButtonActionsContainer = container.locator( '.elementor-editor-container-settings' );
+		const actions = contactButtonActionsContainer.locator( 'li' );
```
Comment (louiswol94) [inline tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:66]: I don't think we need this assertion

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+	}, testInfo ) => {
+		const floatingElPage = new FloatingElementPage( page, testInfo, apiRequests );
+		await floatingElPage.goToFloatingButtonsEmptyPage();
+
+		const addNewButtonId = '#elementor-template-library-add-new';
+		const addNewButton = page.locator( addNewButtonId );
+
+		await test.step( 'Check that buttons and top bar exists', async () => {
+			await expect( addNewButton ).toBeVisible();
+			const topBar = page.locator( '#e-admin-top-bar-root' );
```
Comment (louiswol94) [inline tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:137]: I don't think we need to check of the Top Bar exists

**Review by louiswol94 — CHANGES_REQUESTED** [review]

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+			'.elementor-control-section_custom_attributes_pro',
+		];
+
+		controlsToCheck.forEach( async ( control ) => {
+			const element = elementorControlsContainer.locator( control );
+			await expect( element ).toBeVisible();
+		} );
+
+		const sections = elementorControlsContainer.locator( '.elementor-control-type-section' );
+		await expect( sections ).toHaveCount( 5 );
```
[the remark this answers is not in the document]
Comment (nicoladj77) [reply tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:57]: @louiswol94 this is to check that the advanced section doesn't have the extra sections from the common controls, which has happened a lot and for which we have code

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+		await page.waitForURL( '/wp-admin/edit.php?post_type=e-floating-buttons' );
+
+		const context = await browser.newContext();
+		const newPage = await context.newPage();
+		await newPage.goto( 'https://example.com' );
+		await newPage.goto( '/' );
+		const floatingElement = newPage.locator( '.e-contact-buttons' );
+		await expect( floatingElement ).toBeVisible();
+
+		expect( await floatingElement.screenshot() ).toMatchSnapshot();
```
[the remark this answers is not in the document]
Comment (nicoladj77) [reply tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:91]: @louiswol94 no used my locally created one, how do I download it from CI?

File: tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+		await this.page.locator( '#doaction' ).click();
+		await this.page.waitForURL( '/wp-admin/edit.php?post_type=e-floating-buttons&paged=1' );
+	}
+
+	async createFloatingButtonWithAPI() {
+		const request: APIRequestContext = this.page.context().request,
+			postDataInitial = {
+				title: 'Playwright Test Floating Element - Uninitialized',
+				status: 'publish',
+				content: '<button aria-controls="e-contact-buttons__content-wrapper" aria-label="Close Floating Element" type="button">\n' +
```
[the remark this answers is not in the document]
Comment (nicoladj77) [reply tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts:31]: @louiswol94 I think we do, not 100% sure though. it is present when inserted from remote.

File: modules/floating-buttons/module.php [hunk 7]
```diff
@@ -73,6 +73,33 @@ private function register_admin_menu_legacy( Admin_Menu_Manager $admin_menu ) {
 	public function __construct() {
 		parent::__construct();
 
+		if ( isset( $_GET['playwright_testing'] ) ) {
```
[the remark this answers is not in the document]
Comment (nicoladj77) [reply modules/floating-buttons/module.php:76]: @louiswol94 should I always register the meta? because without registering the meta I can't create the post through Rest endpoints, unless I create a custom endpoint

File: modules/floating-buttons/assets/js/shared/frontend/handlers/click-tracking.js (hunk 1 above)
Comment (nicoladj77) [reply modules/floating-buttons/assets/js/shared/frontend/handlers/click-tracking.js:69]: @louiswol94 I could click 10 times, but TBH 10 is an arbitrary number we choose, not sure what makes more sense here. `beforeunload` doesn't seem to fire even if I read that it should fire.

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+		await expect( floatingElementTitle ).toHaveText( `Playwright Test Page #${ editor.postId }` );
+		await floatingElementTitle.hover();
+		const setAsEntireSite = page.locator( '.row-actions .set_as_entire_site' );
+		await expect( setAsEntireSite ).toBeVisible();
+		await setAsEntireSite.click();
+		await page.waitForURL( '/wp-admin/edit.php?post_type=e-floating-buttons' );
+
+		const context = await browser.newContext();
+		const newPage = await context.newPage();
+		await newPage.goto( 'https://example.com' );
```
[the remark this answers is not in the document]
Comment (nicoladj77) [reply tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:86]: @louiswol94 sorry missed it

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts (hunk 2 above)
Comment (nicoladj77) [reply tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:66]: @louiswol94 I think we do because this tests that only the Delete Container button exists, we have code added for that part, to remove the other two buttons.

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts (hunk 8 above)
[the remark this answers is not in the document]
Comment (nicoladj77) [reply tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts]: removed

File: tests/playwright/assets/api-requests.ts [hunk 9]
```diff
@@ -33,6 +33,28 @@ export default class ApiRequests {
 		return id;
 	}
 
+	public async createWithREST( request: APIRequestContext, entity: string, data: PostForREST ) {
```
[the remark this answers is not in the document]
Comment (nicoladj77) [reply tests/playwright/assets/api-requests.ts:36]: @louiswol94 the original create uses the rest api in a weird way, I mean it calls the index.php file with the rest route. This has never worked with creating meta and I tested it a lot. My approach actually calls REST routes.
I have not changed the original method to avoid breaking something, it looked safer to me to just create a new method, let me know ehat you think

File: tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+					_elementor_controls_usage: 'a:2:{s:15:"contact-buttons";a:3:{s:5:"count";i:1;s:15:"control_percent";i:1;s:8:"controls";a:1:{s:7:"content";a:2:{s:15:"top_bar_section";a:2:{s:13:"top_bar_title";i:1;s:13:"top_bar_image";i:1;}s:22:"message_bubble_section";a:1:{s:19:"message_bubble_name";i:1;}}}}s:9:"container";a:3:{s:5:"count";i:1;s:15:"control_percent";i:0;s:8:"controls";a:0:{}}}',
+				},
+			},
+			postId = await this.apiRequests.createWithREST( request, 'e-floating-buttons', postDataInitial ),
+			postDataUpdated = {
+				title: `Playwright Test Page #${ postId }`,
+			};
+
+		await this.apiRequests.create( request, `e-floating-buttons/${ postId }`, postDataUpdated );
```
[the remark this answers is not in the document]
Comment (nicoladj77) [reply tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts:60]: @louiswol94 I copied the approach from other tests, it is good for checking names, otherwise they all have the same name. in this way you mark the post you created with the ID. I can also remove this of course.

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+test.describe( 'Verify floating buttons editor, admin page and front page behavior', () => {
+	test.afterEach( async ( { browser, apiRequests }, testInfo ) => {
+		const context = await browser.newContext();
+		const page = await context.newPage();
+		const floatingElPage = new FloatingElementPage( page, testInfo, apiRequests );
+		await floatingElPage.deleteAllFloatingButtons();
+		await page.close();
+	} );
+
+	test( 'Verify editor behavior by creating post through API and also FE behavior', async (
```
Comment (Svitlana-Dykun) [inline tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:15]: This test looks quite large and should be broken down into several tests with more test steps inside. 
It's better to divide it into logical parts and use test.step for every part.

File: tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (hein-obox) [reply tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts:31]: @nicoladj77 @louiswol94 

Maybe we can import this data from a JSON file, to keep this test more readable?

File: tests/playwright/assets/api-requests.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (hein-obox) [reply tests/playwright/assets/api-requests.ts:36]: @nicoladj77 @louiswol94 
Nice work

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts [hunk 12]
```diff
@@ -0,0 +1,182 @@
...
+		const editor = await floatingElPage.createFloatingButtonWithAPI();
+		await floatingElPage.waitForPanel();
+		const navigatorId = 'header [aria-label="Structure"]';
+		const navigator = page.locator( navigatorId );
+		// We need to check that the normal elements are not present, that the navigator is visible and the title is correct.
+		await expect( navigator ).toBeHidden();
+		const addButton = page.locator( 'header [aria-label="Add Element"]' );
+		await expect( addButton ).toBeHidden();
+		const panelTitle = page.locator( '#elementor-panel-header-title' );
+		await expect( panelTitle ).toHaveText( 'Edit Single Chat' );
```
Comment (Svitlana-Dykun) [inline tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:31]:
```suggestion
await test.step( 'define name', async () => {
	await expect( page.locator( 'header [aria-label="Structure"]' ) ).toBeHidden();
	await expect( page.locator( 'header [aria-label="Add Element"]' ) ).toBeHidden();
	await expect( page.locator( '#elementor-panel-header-title' ) ).toHaveText( 'Edit Single Chat' );
} );
```

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts (hunk 11 above)
Comment (Svitlana-Dykun) [reply tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:15]: Please see an example for lines 24-31

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+
+		// Check that we are seeing our custom advanced tab and that the normal advanced tab is hidden.
+		const navigationContainer = page.locator( '.elementor-panel-navigation' );
+		const buttons = navigationContainer.locator( 'button' );
+		const visibleButtons = buttons.filter( { has: page.locator( ':visible' ) } );
+		await expect( visibleButtons ).toHaveCount( 3 );
```
Comment (Svitlana-Dykun) [inline tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:37]:
```suggestion
		await expect(page.locator('.elementor-panel-navigation button:visible')).toHaveCount(3);
```

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts [hunk 14]
```diff
@@ -0,0 +1,182 @@
...
+		const advancedSettingsCustom = page.locator( '.elementor-tab-control-advanced-tab-floating-buttons' );
+		await expect( advancedSettingsCustom ).toBeVisible();
+		await advancedSettingsCustom.click();
```
Comment (Svitlana-Dykun) [inline tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:40]: It is better to avoid any actions after the assert. It is better to perform a new action in another step.

**Review by Svitlana-Dykun — CHANGES_REQUESTED** [review]

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (louiswol94) [reply tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:90]: It should be in the artifact: https://github.com/elementor/elementor/actions/runs/10558351983/artifacts/1854090456

File: tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (louiswol94) [reply tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts:60]: Oh ok cool!

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (louiswol94) [reply tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:57]: Ok cool. I get you. Let's monitor it.

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts (hunk 2 above)
Comment (louiswol94) [reply tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:66]: Ah ok!

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts (hunk 3 above)
Comment (louiswol94) [reply tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:136]: I think the Top Bar module itself will check if the Top Bar exists in their own tests. You should probably only check if your actions exist or not. WDYT?

**Comment by ronenelementor** [comment]
I personally find the test to be very long and hard to follow, (tests might also fail due to timeout), so I would split it into smaller test-units. I would also like to have some more blank lines and unified consts just for readability but this is only my personal preference (we mostly put a blank lines before and after scopes and variable defining). Maybe try to use shared logic in you helper file to reduce amount of code in the test file?

File: modules/floating-buttons/module.php (hunk 7 above)
[the remark this answers is not in the document]
Comment (louiswol94) [reply modules/floating-buttons/module.php:76]: Can you try `meta_input`?

File: tests/playwright/assets/api-requests.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (louiswol94) [reply tests/playwright/assets/api-requests.ts:36]: @nicoladj77 I think it is dependent on your "permalink structure"? Can you share your code for trying to add meta with the `index.php` route?

I imagine it would be something like this:

```
const response = await request.post(`${ this.baseUrl }/index.php`, {
    params: { 
        rest_route: `/wp/v2/${ entity }`
    },
    headers: {
        'X-WP-Nonce': this.nonce,
    },
    data: {
        title: "Your Post Title",
        content: "This is the content of the custom post type.",
        status: "publish",
        meta_input: {
            your_meta_key: "Your meta value"
        }
    }
});
```

**Comment by nicoladj77** [comment]
@Svitlana-Dykun @hein-obox @louiswol94 I refactored code to make it clearer, let me know what you think.

File: tests/playwright/assets/api-requests.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (nicoladj77) [reply tests/playwright/assets/api-requests.ts:36]: @louiswol94 it was the exact same code. if you look at the Post type it expects meta as a string. I changed that to object but meta was nto arriving correctly

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts (hunk 11 above)
Comment (nicoladj77) [reply tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:15]: @Svitlana-Dykun done

File: modules/floating-buttons/module.php (hunk 7 above)
[the remark this answers is not in the document]
Comment (nicoladj77) [reply modules/floating-buttons/module.php:76]: @louiswol94 with meta_input the meta is ignored, it doesn't give an error but the meta is ignored and post is not correctr. from my testing the only way I got this to work is by using the refactored create function. If you want I can remove the extra function and modify the old one, checking that everything works. 
Regarding registering the meta, I couldn't get this to work without registering the meta. 
Could we add this during the test:setup:playwright phase somehow?

[the remark this answers is not in the document]
Comment (louiswol94) [reply modules/floating-buttons/module.php:76]: @nicoladj77 Sorry for the back and forth on this. Maybe we can use `wp import`?

https://github.com/elementor/elementor-pro/blob/d2fdd658f07ceabd2b48b06edae7e20e3ceeb9e8/tests/wp-env/config/import-sample-data.sh#L3

File: modules/floating-buttons/assets/js/shared/frontend/handlers/click-tracking.js (hunk 1 above)
Comment (louiswol94) [reply modules/floating-buttons/assets/js/shared/frontend/handlers/click-tracking.js:69]: @nicoladj77 As I see it, then we don't have to change anything here 🤔

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (louiswol94) [reply tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts]: @nicoladj77 We only need the `-linux` image. You can delete yours.

[the remark this answers is not in the document]
Comment (nicoladj77) [reply tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts]: @louiswol94 so i renamed the local one and it works, I also had to commit the local one

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts (hunk 3 above)
Comment (nicoladj77) [reply tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:157]: @louiswol94 since we are here on the page and the check takes milliseconds I would check it, don't see any downside.

File: modules/floating-buttons/module.php (hunk 7 above)
[the remark this answers is not in the document]
Comment (nicoladj77) [reply modules/floating-buttons/module.php:76]: @louiswol94 yes I can try to use that, should work.

File: tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (nicoladj77) [reply tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts:31]: @hein-obox so I guess we will move to wp import cli command

[the remark this answers is not in the document]
Comment (hein-obox) [reply tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts:31]: @nicoladj77 I am not so happy with 'hardcoding' all the post content. We usually generate the post content with the Elementor editor inside or the test. Or we import a json template. I am not sure if you can imports inside the floating editor as well.

E.g. https://github.com/elementor/elementor-pro/blob/bc0076c52c09ff9117497a5113d9161e2c055210/tests/playwright/sanity/modules/off-canvas/off-canvas-a11y.test.ts#L63

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts [hunk 15]
```diff
@@ -0,0 +1,209 @@
...
+	} );
+
+	test( 'Verify editor behavior by creating post through API and also FE behavior', async (
+		{
+			browser,
+			page,
+			apiRequests,
+		}, testInfo ) => {
+		const floatingElPage = new FloatingElementPage( page, testInfo, apiRequests );
+		const editor = await floatingElPage.createFloatingButtonWithAPI();
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:22]: Are we testing both the API import method and the Elementor Editor method?

I think just testing the Editor is sufficient.

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts [hunk 16]
```diff
@@ -0,0 +1,209 @@
...
+			await setAsEntireSite.click();
+			await page.waitForURL( '/wp-admin/edit.php?post_type=e-floating-buttons' );
+
+			context = await browser.newContext();
+			newPage = await context.newPage();
+			await newPage.goto( '/' );
+			floatingElement = newPage.locator( '.e-contact-buttons' );
+			await expect( floatingElement ).toBeVisible();
+
+			expect( await floatingElement.screenshot(
```
Comment (hein-obox) [inline tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts:104]: I think I would add a few more screenshots, to ensure that the styling works as expected.

File: tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts [hunk 17]
```diff
@@ -0,0 +1,65 @@
+import WpAdminPage from '../../../pages/wp-admin-page';
+import type { APIRequestContext } from '@playwright/test';
+import EditorPage from '../../../pages/editor-page';
+
+export default class FloatingElementPage extends WpAdminPage {
```
Comment (yotamselementor) [inline tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts:5]: Can we remove the inheritance here? It seems to me like 2 separate classes

File: tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (nicoladj77) [reply tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts:31]: @hein-obox so the big problem here is that the floating buttons are imported from the remote library, and the remote library is not available in the test environment as we can't connect. And also importing from the remote library would make the test flaky as it would depend on the remote library always returning the same thing. So either we find a way to mock the remote library call or we need to import it. I am ok with either wp import, which is what I am working on, or using a json template, but I do not know how to generate one.

File: tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts (hunk 17 above)
Comment (nicoladj77) [reply tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts:5]: @yotamselementor in my tests I use lots of methods of the base class, it was to avoid using two objects and reusing code. Inheritance make sense to me here, but let me know what you think

File: modules/floating-buttons/assets/js/shared/frontend/handlers/click-tracking.js (hunk 1 above)
Comment (nicoladj77) [reply modules/floating-buttons/assets/js/shared/frontend/handlers/click-tracking.js]: @louiswol94 rolled back changes

File: tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts (hunk 15 above)
Comment (nicoladj77) [reply tests/playwright/sanity/modules/floating-buttons/floating-elements-admin.test.ts]: @hein-obox floating elements only come through the remote library

File: tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts (hunk 17 above)
Comment (yotamselementor) [reply tests/playwright/sanity/modules/floating-buttons/floating-element-page.ts:4]: @nicoladj77 I would rather have these 2 classes as decoupled from each other as possible

**Review by yotamselementor — DISMISSED** [review]

**Review by hein-obox — DISMISSED** [review]

**Review by louiswol94 — APPROVED** [review]

**Review by Svitlana-Dykun — APPROVED** [review]
